src/AppBundle/Controller/Ndr/NdrController.php

```diff
@@ -248,9 +249,45 @@ public function submitConfirmationAction(Request $request, $ndrId)
 
         $ndrStatus = new NdrStatusService($ndr);
 
+        $form = $this->createForm(FormDir\FeedbackReportType::class, new ModelDir\FeedbackReport());
+
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
+            $feedbackEmail = $this->getMailFactory()->createFeedbackEmail($form->getData(), $this->getUser());
+            $this->getMailSender()->send($feedbackEmail, ['html']);
+
+            return $this->redirect($this->generateUrl('ndr_submit_feedback', ['ndrId' => $ndrId]));
+        }
+
         return [
             'ndr' => $ndr,
             'ndrStatus' => $ndrStatus,
+            'form' => $form->createView(),
+        ];
+    }
+
+    /**
+     * @Route("/ndr/{ndrId}/submit_feedback", name="ndr_submit_feedback")
+     * @Template("AppBundle:Report:Report/submitFeedback.html.twig")
+     */
+    public function submitFeedbackAction($ndrId)
+    {
+        $client = $this->getFirstClient(self::$ndrGroupsForValidation);
+        $ndr = $client->getNdr();
+        if ($ndr->getId() != $ndrId) {
+            throw new \RuntimeException('Not authorised to access this Report');
+        }
+        $ndr->setClient($client);
```

Comment:
why are we setting the client to the ndr? surely its already set and known if the ndr has been submitted?